### PR TITLE
relay: fix `no such file or directory` for start-relay command

### LINKED ISSUE #2136: relay: start-relay failed with `no such file or directory`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
upgrade from v2.0.1 to v2.0.6
start-relay
2. What did you expect to see?
start-relay successfully
3. What did you see instead?
```
flush local meta, Rawcause: open relay-dir/xxx.000001/relay.metayyyy: no such file or directory
```
4. Possible causes  

sync --- with relay
restart worker
sync --- without relay
start-relay

now DM finds the previous relay log, try to recover it
https://github.com/pingcap/dm/blob/e56da7da7d812fcf44867a3cb0c121ae34765c50/relay/relay.go#L216
If the relay log really recoverd, relay meta will be dirty
https://github.com/pingcap/dm/blob/e56da7da7d812fcf44867a3cb0c121ae34765c50/relay/relay.go#L443
if upstream has rotated binlog, relay meta is outdated(checkpoint is bigger than relay file)
https://github.com/pingcap/dm/blob/e56da7da7d812fcf44867a3cb0c121ae34765c50/relay/relay.go#L241
so DM purge relay dir
https://github.com/pingcap/dm/blob/e56da7da7d812fcf44867a3cb0c121ae34765c50/relay/relay.go#L246
then when DM create new uuid dir
https://github.com/pingcap/dm/blob/e56da7da7d812fcf44867a3cb0c121ae34765c50/relay/relay.go#L258
will flush dirty meta before the dir created and than report `no such file or directory`
https://github.com/pingcap/dm/blob/e56da7da7d812fcf44867a3cb0c121ae34765c50/relay/meta.go#L294-L303

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #2136 

### What is changed and how it works?
reset meta after purgeRelayDir like we do in [reSetupMeta](https://github.com/pingcap/dm/blob/9fe475a41369f88692b9ab6d62ed847492918082/relay/relay.go#L647)

### Check List 

Tests 
 - Integration test

### CODE REVIEW

**Comment by GMHDBJD** [comment]
PTAL, ebe19ea reproduce the bug, 02ca0ab fix the bug.

File: relay/relay.go [hunk 1]
```diff
@@ -243,11 +243,13 @@ func (r *Relay) process(ctx context.Context) error {
 		}
 
 		if isRelayMetaOutdated {
+			uuidWithSuffix := r.meta.UUID() // only change after switch
```
Comment (Ehco1996) [inline relay/relay.go:246]: why move this line ?

File: relay/relay.go [hunk 2]
```diff
@@ -243,11 +243,13 @@ func (r *Relay) process(ctx context.Context) error {
...
 			err2 = r.PurgeRelayDir()
 			if err2 != nil {
 				return err2
 			}
-			uuidWithSuffix := r.meta.UUID() // only change after switch
+			r.ResetMeta()
+
 			uuid, _, err3 := utils.ParseSuffixForUUID(uuidWithSuffix)
 			if err3 != nil {
 				r.logger.Error("parse suffix for UUID when relay meta oudated", zap.String("UUID", uuidWithSuffix), zap.Error(err))
```
Comment (Ehco1996) [inline relay/relay.go:255]: outdated

**Review by Ehco1996 — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**